wrapper/src/main/java/software/amazon/jdbc/MonitorPluginService.java

```diff
@@ -0,0 +1,795 @@
+    }
+  }
+
+  @Override
+  public boolean forceRefreshHostList(final boolean shouldVerifyWriter, final long timeoutMs)
+      throws SQLException {
+
+    final HostListProvider hostListProvider = this.getHostListProvider();
+    if (!(hostListProvider instanceof BlockingHostListProvider)) {
+      throw new UnsupportedOperationException(
+          Messages.get("PluginServiceImpl.requiredBlockingHostListProvider",
+              new Object[] {hostListProvider.getClass().getName()}));
+    }
+
+    try {
+      final List<HostSpec> updatedHostList =
+          ((BlockingHostListProvider) hostListProvider).forceRefresh(shouldVerifyWriter, timeoutMs);
+      if (updatedHostList != null) {
+        updateHostAvailability(updatedHostList);
+        setNodeList(this.allHosts, updatedHostList);
+        return true;
+      }
+    } catch (TimeoutException ex) {
+      // do nothing.
+      LOGGER.finest(Messages.get("PluginServiceImpl.forceRefreshTimeout", new Object[] {timeoutMs}));
+    }
+    return false;
+  }
+
+  void setNodeList(@Nullable final List<HostSpec> oldHosts,
+      @Nullable final List<HostSpec> newHosts) {
+
+    final Map<String, HostSpec> oldHostMap = oldHosts == null
+        ? new HashMap<>()
+        : oldHosts.stream().collect(Collectors.toMap(HostSpec::getUrl, (value) -> value));
+
+    final Map<String, HostSpec> newHostMap = newHosts == null
+        ? new HashMap<>()
+        : newHosts.stream().collect(Collectors.toMap(HostSpec::getUrl, (value) -> value));
+
+    final Map<String, EnumSet<NodeChangeOptions>> changes = new HashMap<>();
+
+    for (final Map.Entry<String, HostSpec> entry : oldHostMap.entrySet()) {
+      final HostSpec correspondingNewHost = newHostMap.get(entry.getKey());
+      if (correspondingNewHost == null) {
+        // host deleted
+        changes.put(entry.getKey(), EnumSet.of(NodeChangeOptions.NODE_DELETED));
+      } else {
+        // host maybe changed
+        final EnumSet<NodeChangeOptions> hostChanges = compare(entry.getValue(), correspondingNewHost);
+        if (!hostChanges.isEmpty()) {
+          changes.put(entry.getKey(), hostChanges);
+        }
+      }
+    }
+
+    for (final Map.Entry<String, HostSpec> entry : newHostMap.entrySet()) {
+      if (!oldHostMap.containsKey(entry.getKey())) {
+        // host added
+        changes.put(entry.getKey(), EnumSet.of(NodeChangeOptions.NODE_ADDED));
+      }
+    }
+
+    if (!changes.isEmpty()) {
+      this.allHosts = newHosts != null ? newHosts : new ArrayList<>();
+      this.pluginManager.notifyNodeListChanged(changes);
+    }
+  }
+
+  @Override
+  public boolean isStaticHostListProvider() {
+    return this.getHostListProvider() instanceof StaticHostListProvider;
+  }
+
+  @Override
+  public void setHostListProvider(final HostListProvider hostListProvider) {
+    this.hostListProvider = hostListProvider;
+  }
+
+  @Override
+  public Connection connect(final HostSpec hostSpec, final Properties props) throws SQLException {
```

Comment:
I believe we need to raise an exception here. We don't want to support this pipeline.